### PR TITLE
Add symfony 4 support

### LINKED ISSUE #36: Add symfony4 support

### PR DESCRIPTION
**Pull request by johnsudaar**
Fix #36

### CODE REVIEW

File: frameworks/symfony [hunk 1]
```diff
@@ -49,19 +59,29 @@ case "$1" in
...
+        fi
         ;;
     end)
         BUILD_DIR="$1"
         CACHE_DIR="$2"
         APP_DIR="/app"
 
-        status "Warmuping cache"
-        rm -rf app/cache/*
+        status "Warmping cache"
```
Comment (yannski) [inline frameworks/symfony]: typo here

File: frameworks/symfony [hunk 2]
```diff
@@ -49,19 +59,29 @@ case "$1" in
...
 
         workdir=$(get_workdir)
         console=$(get_console)
-        php $console cache:warmup --env=prod --no-debug | indent
+        rm -rf $workdir/cache/*
+        if is_symfony4 ; then
+          cd bin
+          php console cache:warmup --env=prod --no-debug | indent
```
Comment (yannski) [inline frameworks/symfony]: shouldn't it be $console?

File: frameworks/symfony [hunk 3]
```diff
@@ -49,19 +59,29 @@ case "$1" in
...
 
-        status "Warmuping cache"
-        rm -rf app/cache/*
+        status "Warmping cache"
 
         workdir=$(get_workdir)
         console=$(get_console)
-        php $console cache:warmup --env=prod --no-debug | indent
+        rm -rf $workdir/cache/*
+        if is_symfony4 ; then
```
Comment (Soulou) [inline frameworks/symfony]: Why this? the `$console` variable should contain `bin/console` or `app/console` according to the version of symfony

File: frameworks/symfony [hunk 4]
```diff
@@ -20,14 +24,20 @@ function is_symfony3() {
     test -e "bin/console"
 }
 
+function is_symfony4() {
```
Comment (EtienneM) [inline frameworks/symfony]: There is not much difference between `is_symfony4` and `requires_symfony_flex`. Isn't it possible to factorize a bit?

File: frameworks/symfony (hunk 3 above)
Comment (johnsudaar) [reply frameworks/symfony]: Well yep but with `php $console [...]` the script return:

```
2018-01-08T15:59:36+00:00 [error] Error thrown while running command "'bin/console' 'cache:warmup' --env=prod --no-debug". Message: "Command "bin/console" is not defined."                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                                                              
  Command "bin/console" is not defined.                                                                                                                                                                                                                                                                                                                                                       
```

They must use $0 wrongly

File: frameworks/symfony (hunk 2 above)
Comment (johnsudaar) [reply frameworks/symfony]: See my answer to @Soulou

File: frameworks/symfony (hunk 1 above)
Comment (johnsudaar) [reply frameworks/symfony]: ✅

File: frameworks/symfony (hunk 4 above)
Comment (johnsudaar) [reply frameworks/symfony]: ✅

File: frameworks/symfony (hunk 3 above)
Comment (Soulou) [reply frameworks/symfony]: Please let a comment then

Comment (johnsudaar) [reply frameworks/symfony]: Already done

**Pull request merged**